numerapi/numerapi.py

```diff
@@ -753,9 +765,9 @@ def upload_predictions(self, file_path: str, tournament: int = 8,
 
         # get compute id if available and pass it along
         headers = {"x_compute_id": os.getenv("NUMERAI_COMPUTE_ID")}
-        with open(file_path, 'rb') as fh:
+        with open(file_path, 'rb') if df is None else buffer_csv as fh:
             requests.put(
-                submission_auth['url'], data=fh.read(), headers=headers)
+                submission_auth['url'], data=fh.read() if df is None else fh.getvalue(), headers=headers)
```

Comment:
Add a `buffer_csv.seek(0)` right after writing the csv to it. Then you can just call `read` on it and this line of code doesn't need the `if`